include/crocoddyl/core/integrator/euler.hpp

```diff
@@ -28,7 +28,7 @@ class IntegratedActionModelEuler : public ActionModelAbstract {
 
   const boost::shared_ptr<DifferentialActionModelAbstract>& get_differential() const;
   const double& get_dt() const;
-  void set_dt(double dt);
+  void set_dt(const double& dt);
```

Comment:
For primitive types there is no benefit of using const-ref. The cost of creating a reference and the size of it is effectively the same as passing by value and thus creating a copy.

Reply:
I know about the primitives, however I believe it's a good rule of thumb to use it in these cases.

Reply:
@jcarpent, @nim65s and @olivier-stasse 

Should we use only object for primitive types (double, int, bool)?

Reply:
I'd say no, we should just keep `(double dt)` for primitives types, but I don't have references about this pattern right now.

Reply:
@nim65s Sorry I didn't get your point. Do you mean "use `double`, instead of `const double&`"?

According to this link (https://stackoverflow.com/a/14013189), the size of a `double` are bigger than reference. So, for `double`, it could be better to use `const&` if extra aliasing it's not an issue (which I believe for these functions the answer is "no issue"`)

Reply:
I'd say: If it's a primitive, pass by value. Only pass by const-ref for larger objects. So for here `(double dt)`

From the same link:
> pass-by-value avoids aliasing, allowing the optimizer to really do its thing

Reply:
@cmastalli : yes, I mean "use `double`, instead of `const double&`", but if you want to benchmark it, and if that benchmark shows a good (performances gains / complexity loss) ratio, we can discuss it.

Reply:
In my case, I would just use const double to say to the compiler that this quantity won't be modified ;). But I do agree with @wxmerkt, this is not necessary at first glance, except when you use AutoDiiff frameworks!

Reply:
@jcarpent thanks for the feedback.